p2p/pubsub/metric.go

```diff
@@ -0,0 +1,86 @@
+package pubsub
```

Comment:
if we are designing a new metrics system, we'd better use a global metrics system that can be added for each layer.  So that we don't have to add metrics system for each layer.

Reply:
I am thinking to add a new metrics system using Prometheus https://prometheus.io/docs/guides/go-application/

Reply:
Oh, yea. This can be much greater than the simple metric we implement ourselves. Let me think it over and maybe we can do in another thread.